requirements.txt

```diff
@@ -3,28 +3,28 @@
 # Docs
 mkdocs==1.6.0
 mkdocs-autorefs==1.0.1
-mkdocs-material==9.5.27
+mkdocs-material==9.5.30
 mkdocs-material-extensions==1.3.1
-mkdocstrings[python-legacy]==0.25.1
+mkdocstrings[python-legacy]==0.25.2
 jinja2==3.1.4
 
 # Packaging
 build==1.2.1
 twine==5.1.1
 
 # Tests & Linting
-coverage[toml]==7.5.4
-ruff==0.5.0
-mypy==1.10.1
+coverage[toml]==7.6.0
+ruff==0.5.5
+mypy==1.11.1
 trio-typing==0.10.0
-pytest==8.2.2
+pytest==8.3.2
 pytest-httpbin==2.0.0
 pytest-trio==0.8.0
 werkzeug<3.1  # See: https://github.com/psf/httpbin/issues/35
 
 # Benchmarking and profiling
-uvicorn==0.30.1
-aiohttp==3.9.5
+uvicorn==0.30.4
+aiohttp==3.10.0
 urllib3==2.2.2
-matplotlib==3.7.5
+matplotlib==3.9.1
```

Comment:
This upgrade is breaking the builds.

I'd suggest that we drop the dependencies introduced in PR #923 out of the `requirements.txt`. We don't really want to be installing these during every CI run. Perhaps have a `/benchmark` directory with dependencies and the `client`/`server` scripts?